lms/djangoapps/instructor/tests/test_registration_codes.py

```diff
@@ -0,0 +1,280 @@
+)
+from student.models import CourseEnrollment
+from student.roles import CourseSalesAdminRole
+from nose.plugins.attrib import attr
+import json
+from student.tests.factories import UserFactory, CourseModeFactory
+from django.core.urlresolvers import reverse
+from django.test.utils import override_settings
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+
+
+@attr('shard_1')
+@override_settings(REGISTRATION_CODE_LENGTH=8)
+class TestCourseRegistrationCodeStatus(ModuleStoreTestCase):
+    """
+    Test registration code status.
+    """
+
+    def setUp(self):
+        """
+        Fixtures.
+        """
+        super(TestCourseRegistrationCodeStatus, self).setUp()
+        self.course = CourseFactory.create()
+        CourseModeFactory.create(course_id=self.course.id, min_price=50)
+        self.instructor = InstructorFactory(course_key=self.course.id)
+        self.client.login(username=self.instructor.username, password='test')
+        CourseSalesAdminRole(self.course.id).add_users(self.instructor)
+
+        # create testing invoice
+        self.sale_invoice = Invoice.objects.create(
+            total_amount=1234.32, company_name='Test1', company_contact_name='TestName',
+            company_contact_email='Test@company.com', recipient_name='Testw', recipient_email='test1@test.com',
+            customer_reference_number='2Fwe23S', internal_reference="A", course_id=self.course.id, is_valid=True
+        )
+        self.invoice_item = CourseRegistrationCodeInvoiceItem.objects.create(
+            invoice=self.sale_invoice,
+            qty=1,
+            unit_price=1234.32,
+            course_id=self.course.id
+        )
+        self.lookup_code_url = reverse('look_up_registration_code',
+                                       kwargs={'course_id': unicode(self.course.id)})
+
+        self.registration_code_detail_url = reverse('registration_code_details',
+                                                    kwargs={'course_id': unicode(self.course.id)})
+
+        url = reverse('generate_registration_codes',
+                      kwargs={'course_id': self.course.id.to_deprecated_string()})
+
+        data = {
+            'total_registration_codes': 12, 'company_name': 'Test Group', 'company_contact_name': 'Test@company.com',
+            'company_contact_email': 'Test@company.com', 'unit_price': 122.45, 'recipient_name': 'Test123',
+            'recipient_email': 'test@123.com', 'address_line_1': 'Portland Street',
+            'address_line_2': '', 'address_line_3': '', 'city': '', 'state': '', 'zip': '', 'country': '',
+            'customer_reference_number': '123A23F', 'internal_reference': '', 'invoice': ''
+        }
+
+        response = self.client.post(url, data, **{'HTTP_HOST': 'localhost'})
+        self.assertEqual(response.status_code, 200, response.content)
+
+    def test_look_up_invalid_registration_code(self):
+        """
+        test lookup for the invalid registration code
```

Comment:
As written, you've simply repeated the function name. This comment can be improved by also describing the expected behavior. For example: `Verify the view returns HTTP status 400 if an invalid registration code is passed. Also, verify the data returned includes a message indicating the error, and the is_registration_code_valid is set to False.